codalab/worker/file_util.py

```diff
@@ -211,31 +211,12 @@ def gzip_file(file_path):
     """
     Returns a file-like object containing the gzipped version of the given file.
     """
-    BUFFER_SIZE = 100 * 1024 * 1024  # Zip in chunks of 100MB
-
-    class GzipStream:
-        def __init__(self, fileobj):
-            self.__input = fileobj
-            self.__buffer = BytesBuffer()
-            self.__gzip = gzip.GzipFile(None, mode='wb', fileobj=self.__buffer)
-
-        def read(self, size=-1):
-            while size < 0 or len(self.__buffer) < size:
-                s = self.__input.read(BUFFER_SIZE)
-                if not s:
-                    self.__gzip.close()
-                    break
-                self.__gzip.write(s)
-            return self.__buffer.read(size)
-
-        def close(self):
-            self.__input.close()
-
+    args = ['gzip', '-c', '-n', file_path]
```

Comment:
Could you document that it's right now important for gzip to run in a separate process, otherwise things grind to a halt?